### ISSUE TITLE
Camera1 API  - SurfaceTexture - freeze after reopen.

### ISSUE THREAD

**Issue by Mariluk** [flutter/flutter#28458]
Hello,

When I repoen my camera view with Camera 1API and recreated `SurfaceTexture`. Handler in `SurfaceTexture` handle message only once and texture freezed. 

In `SurfaceTexture` 

```java
public void setOnFrameAvailableListener(@Nullable final OnFrameAvailableListener listener, @Nullable Handler handler) {
       if (listener != null) {
            // Although we claim the thread is arbitrary, earlier implementation would
            // prefer to send the callback on the creating looper or the main looper
            // so we preserve this behavior here.
            Looper looper = handler != null ? handler.getLooper() :
                    mCreatorLooper != null ? mCreatorLooper : Looper.getMainLooper();
            mOnFrameAvailableHandler = new Handler(looper, null, true /*async*/) {
                @Override
                public void handleMessage(Message msg) {
                    listener.onFrameAvailable(SurfaceTexture.this);
                }
            };
        } else {
            mOnFrameAvailableHandler = null;
        }
    }
```
Camera2 API works fine. 

flutter doctor:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.2.3-pre.32, on Linux, locale cs_CZ.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Android Studio (version 3.3)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by Mariluk** [comment]
I can have problem with my camera code. I'm not sure. 

Create Camera

 ```
private Camera1(final Camera camera, final CameraStateListener stateListener, FlutterView.SurfaceTextureEntry textureEntry, Camera.Size captureSize){
        this.camera = camera;
        this.cameraStateListener = stateListener;

        this.surfaceTexture = textureEntry.surfaceTexture();
        surfaceTexture.setDefaultBufferSize(captureSize.width, captureSize.height);

        this.qrReader = new MultiFormatReader();

        HashMap<DecodeHintType, Object> hints = new HashMap<>();
        hints.put(DecodeHintType.POSSIBLE_FORMATS, Arrays.asList(BarcodeFormat.QR_CODE, BarcodeFormat.AZTEC, BarcodeFormat.PDF_417, BarcodeFormat.DATA_MATRIX));
        this.qrReader.setHints(hints);
    }

    public static Camera1 createCameraInstance(FlutterView flutterView, PreviewQuality previewQuality, final MethodChannel.Result result, int captureWidth, int captureHeight, int rotation, final CameraStateListener cameraStateListener, final boolean legacy) throws Exception {
        int cameraId = -1;

        for(int i = 0; i < Camera.getNumberOfCameras(); i++){
            Camera.CameraInfo info = new Camera.CameraInfo();
            Camera.getCameraInfo(i, info);
            cameraId = i;
            if(info.facing == Camera.CameraInfo.CAMERA_FACING_BACK){
                break;
            }
        }


        if(cameraId == -1){
            //TODO replace to camera exception for camera first
            result.error("Camera", "Couldn't find any useable back-facing camera.", null);
            throw new Exception("Couldn't find any useable back-facing camera.");
        }


        FlutterView.SurfaceTextureEntry surfaceTextureEntry = flutterView.createSurfaceTexture();

        Camera camera = Camera.open(cameraId);
        Camera.Parameters parameters = camera.getParameters();


        Camera.Size previewSize = ScannerUtils.getOptimalSize(previewQuality.getPreviewWidth(), previewQuality.getPreviewHeight(), parameters.getSupportedPreviewSizes());
        Camera.Size captureSize = ScannerUtils.getOptimalSize(captureWidth, captureHeight, parameters.getSupportedPreviewSizes());

        HashMap<String, Object> response = new HashMap<>();
        response.put("textureId", cameraId);
        response.put("previewWidth", previewSize.width);
        response.put("previewHeight", previewSize.height);
        response.put("orientation", rotation);
        response.put("legacy", legacy);
        result.success(response);

        if(legacy) {
            parameters.setPreviewFormat(ImageFormat.NV21);
            parameters.setRotation(rotation);
        }

        parameters.setPictureSize(previewSize.width, previewSize.height);
        parameters.setPreviewSize(captureSize.width, captureSize.height);

        if (parameters.getSupportedFocusModes().contains(Camera.Parameters.FOCUS_MODE_CONTINUOUS_VIDEO)) {
            parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_CONTINUOUS_VIDEO);
        }
        camera.setParameters(parameters);

        return new Camera1(camera, cameraStateListener, surfaceTextureEntry, captureSize);
    }

```
Stop preview. 

```
@Override
    public void stopPreview() throws Exception {
        camera.setPreviewTexture(null);
        camera.stopPreview();
    }
```

Dispose method.

```
 @Override
    public void dispose() {
        if(camera != null){
            camera.setPreviewCallback(null);
            camera.release();
            camera = null;
        }
    }
```

**Comment by Mariluk** [comment]
Tested on Android 8.1.0 (API 27) and Android 4.4.2 (API 19)

**Comment by TahaTesser** [comment]
Hi @Mariluk 
if you're still experiencing this problem, please create a new issue and provide the information required in the template.
Thank you
